src/record.py

```diff
@@ -0,0 +1,44 @@
+"""Record class."""
+from parse import get_parser, get_card
+from extract import read_to_dict
+from standardize import standardizer, strip_payment
+from common import flip_spending_sign
+import pandas as pd
+import matplotlib.pyplot as plt
+
+
+class CardRecord(object):
+    def __init__(self, filename):
+        self.filename = filename
+        self.card = get_card(filename)
+        self.parser = get_parser(filename)
+
+    def read_expenses(self):
+        records = read_to_dict(self.filename)
+        records = [standardizer(self.parser(record)) for record in records]
+        self.records = flip_spending_sign(records)
+
+    def to_df(self):
+        return pd.DataFrame(self.records)
+
+    @property
+    def category_colors(self):
+        categories = self.to_df()["category"].unique()
+        colors = plt.get_cmap("tab10").colors
+        cmap = {cat: colors[i] for i, cat in enumerate(categories)}
+        return pd.Series(cmap)
+
+    def plot_categories(self):
+        records_df = pd.DataFrame(strip_payment(self.records))
```

Comment:
this seems used twice, should it just be in init?

Reply:
Records could also just be a dataframe I think.